src/rv32_constopt.c

```diff
@@ -815,9 +812,9 @@ CONSTOPT(clui, {
  */
 CONSTOPT(csrli, {
     if (constopt_info->is_constant[ir->rs1]) {
-        constopt_info->is_constant[ir->rs1] = true;
         ir->imm = constopt_info->const_val[ir->rs1] >> ir->shamt;
-        constopt_info->const_val[ir->rd] = ir->imm;
+        constopt_info->const_val[ir->rs1] = ir->imm;
+        ir->rd = ir->rs1;
         ir->opcode = rv_insn_clui;
         ir->impl = dispatch_table[ir->opcode];
     } else
```

Comment:
How about those else cases here? A statement like `constopt_info->is_constant[ir->rs1] = false` here should also be redundant.

Reply:
Yes, it is redundant, I'll remove it.